vars/piperPipelineStageInit.groovy

```diff
@@ -133,42 +133,35 @@ private void initStashConfiguration (script, config) {
 
 private void setGitUrlsOnCommonPipelineEnvironment(script, String gitUrl) {
 
-    def gitPath = ''
-    if (gitUrl.startsWith('http')) {
-        def httpPattern = /(https?):\/\/([^:\/]+)(?:[:\d\/]*)(.*)/
-        def gitMatcher = gitUrl =~ httpPattern
-        if (!gitMatcher.hasGroup() && gitMatcher.groupCount() != 3) return
-        script.commonPipelineEnvironment.setGitSshUrl("git@${gitMatcher[0][2]}:${gitMatcher[0][3]}")
-        gitPath = gitMatcher[0][3]
+    def urlMatcher = gitUrl =~ /^((http|https|git|ssh):\/\/)?((.*)@)?([^:\/]+)(:([\d]*))?(\/?(.*))$/
```

Comment:
We could instead use `new URI(gitUrl)` here.

Reply:
I played around, but I was not satisfied. After so much time I don't remember the details. Was it the port which was -1, was it an issue with the git protocol which was not recognized ... something like that. In the light of migrating to go I guess having that regex is not that bad.